### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by leandro2219** [kenhyuwa/litepie-datepicker#39]
Hi, I am using this plugin in order to select a date and a range, but I need to get when use select a date (or range) and when It clicks on apply and cancel button. Is this possible?

Thanks.

**Comment by kenhyuwa** [comment]
do you mean props `auto-apply`?

**Comment by leandro2219** [comment]
Hi, I mean, if there is a way than this component emit an event when user select a date or when clicks on Apply or Cancel Button.

Ej:
@click="$emit('select', $event)"

Thanks.

**Comment by sdoerger** [comment]
Indeed, I mean, that is mostly the point to use a datepicker. Yes, I get the dates in my component, but I need to the dates to its parent component.

**Comment by Emilien-Lambert** [comment]
I have the same need, has anyone found a solution?

**Comment by mustafakucuk** [comment]
Hi,

Maybe you want use this way.

```js
import { ref, watch } from "vue";
const dateValue = ref([]);

watch(
  dateValue,
  (val) => {
    console.log(val);
  },
  { deep: true }
);

```

**Comment by jonhenning** [comment]
I believe this is what you are looking for
`<litepie-datepicker @update:modelValue="$emit('select', $event)"`
